File: examples/request.ts
import { encodeRequest } from "../src/protocol";

const timer = setTimeout(() => {
  console.error("No reply; is bun run bridge:start running?");
  socket.close();
  process.exitCode = 1;
}, 2500);
const socket = await Bun.udpSocket({
  hostname: "127.0.0.1",
  connect: { hostname: "127.0.0.1", port: 18765 },
  socket: {
    data(socket, packet) {
      console.log(packet.toString());
      clearTimeout(timer);
      socket.close();
    },
  },
});
socket.send(
  encodeRequest({
    id: "1",
    context: "我用Python写了一个",
    input: "igui",
    candidates: ["城市", "程式", "诚实"],
  }),
);

File: examples/request.py
"""Send one example to a running local Riwen bridge."""
from pathlib import Path
import socket
import sys
sys.path.insert(0, str(Path(__file__).resolve().parents[1] / "src"))
from riwen.protocol import Request, encode_request

with socket.socket(socket.AF_INET, socket.SOCK_DGRAM) as client:
    client.settimeout(2.5)
    client.connect(("127.0.0.1", 18765))
    client.send(encode_request(Request("1", "我用Python写了一个", "igui", ["城市", "程式", "诚实"])))
    print(client.recv(12000).decode())
